adapters/adkernel/adkernel.go

```diff
@@ -90,54 +96,75 @@ func dispatchImpressions(imps []openrtb2.Imp, impsExt []openrtb_ext.ExtImpAdkern
 	errors := make([]error, 0)
 	for idx := range imps {
 		imp := imps[idx]
-		err := compatImpression(&imp)
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
+		imp.Ext = nil
 		impExt := impsExt[idx]
 		if res[impExt] == nil {
 			res[impExt] = make([]openrtb2.Imp, 0)
 		}
-		res[impExt] = append(res[impExt], imp)
+		if isMultiFormatImp(&imp) {
+			splImps := splitMultiFormatImp(&imp)
+			res[impExt] = append(res[impExt], splImps...)
+		} else {
+			res[impExt] = append(res[impExt], imp)
+		}
 	}
 	return res, errors
 }
 
-// Alter impression info to comply with adkernel platform requirements
-func compatImpression(imp *openrtb2.Imp) error {
-	imp.Ext = nil //do not forward ext to adkernel platform
-	if imp.Banner != nil {
-		return compatBannerImpression(imp)
-	}
+func isMultiFormatImp(imp *openrtb2.Imp) bool {
+	count := 0
 	if imp.Video != nil {
-		return compatVideoImpression(imp)
+		count++
+	}
+	if imp.Audio != nil {
+		count++
+	}
+	if imp.Banner != nil {
+		count++
 	}
 	if imp.Native != nil {
-		return compatNativeImpression(imp)
+		count++
 	}
-	return newBadInputError("Invalid impression")
+	return count > 1
 }
 
-func compatBannerImpression(imp *openrtb2.Imp) error {
-	imp.Audio = nil
-	imp.Video = nil
-	imp.Native = nil
-	return nil
-}
+func splitMultiFormatImp(imp *openrtb2.Imp) []openrtb2.Imp {
+	splitImps := make([]openrtb2.Imp, 0, 4)
+	if imp.Banner != nil {
+		impCopy := *imp
+		impCopy.Video = nil
+		impCopy.Native = nil
+		impCopy.Audio = nil
+		impCopy.ID += mf_suffix_banner
+		splitImps = append(splitImps, impCopy)
+	}
+	if imp.Video != nil {
+		impCopy := *imp
+		impCopy.Banner = nil
+		impCopy.Native = nil
+		impCopy.Audio = nil
+		impCopy.ID += mf_suffix_video
+		splitImps = append(splitImps, impCopy)
+	}
 
-func compatVideoImpression(imp *openrtb2.Imp) error {
-	imp.Banner = nil
-	imp.Audio = nil
-	imp.Native = nil
-	return nil
-}
+	if imp.Native != nil {
+		impCopy := *imp
+		impCopy.Banner = nil
+		impCopy.Video = nil
+		impCopy.Audio = nil
+		impCopy.ID += mf_suffix_native
+		splitImps = append(splitImps, impCopy)
+	}
 
-func compatNativeImpression(imp *openrtb2.Imp) error {
-	imp.Banner = nil
-	imp.Audio = nil
-	imp.Video = nil
-	return nil
+	if imp.Audio != nil {
+		impCopy := *imp
+		impCopy.Banner = nil
+		impCopy.Video = nil
+		impCopy.Native = nil
+		impCopy.ID += mf_suffix_audio
+		splitImps = append(splitImps, impCopy)
```

Comment:
as per `static/bidder-info/adkernel.yaml` audio format is not supported. Should remove `imp.Audio != nil` check or update `static/bidder-info/adkernel.yaml` to support audio format